src/Transformer/DetermineHeroImages.php

```diff
@@ -68,44 +68,46 @@ final class DetermineHeroImages implements Transformer {
 	 * @return void
 	 */
 	public function transform( Document $document, ErrorCollection $errors ) { // phpcs:ignore VariableAnalysis.CodeAnalysis.VariableAnalysis.UnusedVariable
-		$existing_hero_images_count = $document->xpath->query(
-			self::EXISTING_HERO_IMAGES_XPATH_QUERY,
-			$document->body
-		)->length;
-
-		$available_hero_image_slots = max(
-			PreloadHeroImage::DATA_HERO_MAX - $existing_hero_images_count,
-			0
-		);
-
 		$hero_image_elements = [];
 
-		if ( count( $hero_image_elements ) < $available_hero_image_slots ) {
-			$custom_logo = $this->get_custom_logo( $document );
-			if ( null !== $custom_logo ) {
-				$hero_image_elements[] = $custom_logo;
-			}
-		}
+		foreach ( [ 'custom_header', 'custom_logo', 'featured_image', 'cover_blocks' ] as $hero_image_source ) {
+			if ( count( $hero_image_elements ) < PreloadHeroImage::DATA_HERO_MAX ) {
+				$get_candidate_method = "get_{$hero_image_source}";
```

Comment:
Minor thing, but IntelliJ is complaining to me about the methods not being used. Would a `switch()` be better in that case for static analysis?

Reply:
Fixed via https://github.com/ampproject/amp-wp/pull/5934/commits/b685b0e1828fed6a9ce4a99ff474f24ba2dca506